pkg/server/api/entry/v1/service_test.go

```diff
@@ -2893,6 +2896,510 @@ func TestGetAuthorizedEntries(t *testing.T) {
+		},
+		{
+			name: "initial request specifies IDs",
+			steps: []step{
+				{
+					req:  &entryv1.SyncAuthorizedEntriesRequest{Ids: []string{"entry-1"}},
+					err:  "specifying IDs on initial request is not supported",
+					code: codes.InvalidArgument,
+				},
+			},
+			expectLogs: []spiretest.LogEntry{
+				{
+					Level:   logrus.InfoLevel,
+					Message: "API accessed",
+					Data: logrus.Fields{
+						telemetry.Status:        "error",
+						telemetry.Type:          "audit",
+						telemetry.StatusCode:    "InvalidArgument",
+						telemetry.StatusMessage: "specifying IDs on initial request is not supported",
+					},
+				},
+			},
+		},
+	} {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			test := setupServiceTest(t, fakedatastore.New(t))
+			defer func() {
+				test.Cleanup()
+				spiretest.AssertLogs(t, test.logHook.AllEntries(), tt.expectLogs)
+			}()
+
+			test.omitCallerID = tt.omitCallerID
+			test.ef.entries = tt.authorizedEntries
+			test.ef.err = tt.fetcherErr
+
+			ctx, cancel := context.WithCancel(ctx)
+			defer cancel()
+
+			stream, err := test.client.SyncAuthorizedEntries(ctx)
+			require.NoError(t, err)
+
+			for i, step := range tt.steps {
+				t.Logf("stream step: %d", i)
+				if step.req != nil {
+					require.NoError(t, stream.Send(step.req))
+				}
+				resp, err := stream.Recv()
+				if step.err != "" {
+					spiretest.RequireGRPCStatusContains(t, err, step.code, step.err)
+					require.Nil(t, resp)
+					return
+				}
+				require.NoError(t, err)
+				spiretest.AssertProtoEqual(t, step.resp, resp)
+			}
+			require.NoError(t, stream.CloseSend())
+		})
+	}
+}
+
+func FuzzSyncAuthorizedStreams(f *testing.F) {
+	rnd := rand.New(rand.NewSource(time.Now().Unix())) //nolint: gosec // this rand source ok for fuzz tests
+
+	const entryPageSize = 5
+
+	calculatePageCount := func(entries int) int {
+		return (entries + (entryPageSize - 1)) / entryPageSize
+	}
+	recvNoError := func(tb testing.TB, stream entryv1.Entry_SyncAuthorizedEntriesClient) *entryv1.SyncAuthorizedEntriesResponse {
+		resp, err := stream.Recv()
+		require.NoError(tb, err)
+		return resp
+	}
+	recvEOF := func(tb testing.TB, stream entryv1.Entry_SyncAuthorizedEntriesClient) {
+		_, err := stream.Recv()
+		require.True(tb, errors.Is(err, io.EOF))
+	}
+
+	const maxEntries = 40
+	var entries []*types.Entry
+	for i := 0; i < maxEntries; i++ {
+		entries = append(entries, &types.Entry{Id: strconv.Itoa(i), RevisionNumber: 1})
+	}
+
+	// Add some quick boundary conditions in here that will be run
+	// during standard testing.
+	f.Add(0, 0)
+	f.Add(1, 1)
+	f.Add(entryPageSize-1, entryPageSize-1)
+	f.Add(entryPageSize, entryPageSize)
+	f.Add(entryPageSize+1, entryPageSize+1)
+	f.Add(0, maxEntries)
+	f.Add(maxEntries/2, maxEntries)
+	f.Add(maxEntries, maxEntries)
+
+	f.Fuzz(func(t *testing.T, staleEntries, totalEntries int) {
+		if totalEntries < 0 || totalEntries > maxEntries {
+			t.Skip()
+		}
+		if staleEntries < 0 || staleEntries > totalEntries {
```

Comment:
based in boundaries, it is not possible to happen, but just in case, can you add logs or error? to prevent to have invalid tests? (same for both skip)

Reply:
Fuzzing will very much produce combinations that hit these conditions.